mmv1/third_party/terraform/services/discoveryengine/discovery_engine_operation.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    Type: MMv1     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package discoveryengine
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+type DiscoveryEngineOperationWaiter struct {
+	Config    *transport_tpg.Config
+	UserAgent string
+	Project   string
+	tpgresource.CommonOperationWaiter
+}
+
+func (w *DiscoveryEngineOperationWaiter) QueryOp() (interface{}, error) {
+	if w == nil {
+		return nil, fmt.Errorf("cannot query operation, it's unset or nil")
+	}
+
+	opName := w.CommonOperationWaiter.Op.Name
+
+	// Extract location from opName.  Example:
+	// "projects/<project>/locations/<location>/operations/<operation-id>"
+	// Requires error handling if not properly formatted.
+	parts := strings.Split(opName, "/")
+	if len(parts) < 5 || parts[2] != "locations" {
+		return nil, fmt.Errorf("unexpected operation name format: %s", opName) // Handle invalid format appropriately
+	}
+	location := parts[3]
+
+	url := fmt.Sprintf("%s/projects/%s/locations/%s/operations/%s", w.Config.DiscoveryEngineBasePath, w.Project, location, parts[5])
```

Comment:
The DiscoveryEngineBasePath will contain the literal string `{{location}}` so that needs to be replaced. I had been thinking it would make sense to do after building the URL but maybe it would be clearer to do it first?

```suggestion
	basePath := strings.Replace(w.Config.DiscoveryEngineBasePath, "{{location}}", location, 1)
	url := fmt.Sprintf("%s%s", basePath, opName)
```

Reply:
updated, thank you for suggestion